src/foam/parse/DateParser.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.parse;
```

Comment:
Can we move this to a sub package like foam.parse.query, as this is pretty specific to the QueryParser